python/lsst/daf/butler/datastores/chainedDatastore.py

```diff
@@ -629,8 +629,13 @@ def forget(self, refs: Iterable[DatasetRef]) -> None:
         for datastore in tuple(self.datastores):
             datastore.forget(refs)
 
-    def trash(self, ref: DatasetRef, ignore_errors: bool = True) -> None:
-        log.debug("Trashing %s", ref)
+    def trash(self, ref: Union[DatasetRef, Iterable[DatasetRef]], ignore_errors: bool = True) -> None:
+        if isinstance(ref, DatasetRef):
+            ref_label = str(ref)
+        else:
+            ref_label = "bulk datasets"
```

Comment:
Might be nice to put `len(ref)` in here.

Reply:
Wouldn't that be a problem if ref was a generator? I assumed that `len()` would trigger the generator and empty it for later.

Reply:
Oh, good point.  If it's a generator or other lazy iterator `len` wouldn't work at all. Nevermind.